src/components/Auth/Details.tsx

```diff
@@ -1,17 +1,56 @@
+import React from "react"
 import { navigate } from "@reach/router"
+import { getCurrentUser } from "../../auth/AppUser"
+import { Link, graphql, useStaticQuery } from "gatsby"
 import { Auth } from "aws-amplify"
-import { Link } from "gatsby"
-import React from "react"
 import styled from "styled-components"
 import colors from "../../colors"
-import { getCurrentUser } from "../../auth/AppUser"
 import { Container } from "../../components/Primitives"
+import { listLikedContents } from "../../graphql/queries"
+import { API, graphqlOperation } from "aws-amplify"
+import LessonCard from "../../components/lesson/LessonCard"
+import WeekCard from "../week/WeekCard"
+import HabitCard from "../Habit/HabitCard"
+import { ContentfulLesson } from "../../../graphql-types"
 import { H2, H3 } from "../Html/HtmlContent"
 import { Icon } from "../Icons"
 import PageHeader from "../PageHeader"
 import UserHabits from "../user/UserHabits"
+import { useQuery } from "react-query"
 
 const Details = () => {
+  const graphqlData = useStaticQuery(graphql`
+    query {
+      allContentfulHabit {
+        edges {
```

Comment:
use nodes instead of edges, it reduces one level of complexity. In gatsby there's a shorthand called nodes which  returns the nodes directly. Experiment with graphql tool